xdsl/dialects/riscv.py

```diff
@@ -361,6 +361,12 @@ def get_register_constraints(self) -> RegisterConstraints:
     def assembly_line(self) -> str | None:
         raise NotImplementedError()
 
+
+class RISCVCustomFormatOperation(RISCVAsmOperation, ABC):
```

Comment:
Is it intentional that this inherits from `RISCVAsmOperation`? Your description makes it sound like these should be disjoint

Reply:
Ah yes I'll remove this, thank you